plugins/woocommerce-admin/client/products/layout/woo-product-field-item.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Slot, Fill } from '@wordpress/components';
+import { cloneElement, isValidElement } from '@wordpress/element';
+import { snakeCase } from 'lodash';
+
+// TODO: move this to a published JS package once ready.
+
+/**
+ * Ordered product field item.
+ *
+ * @param {Node}   children - Node children.
+ * @param {number} order    - Node order.
+ * @param {Array}  props    - Fill props.
+ * @return {Node} Node.
+ */
+const createOrderedChildren = (
+	children: React.ReactNode,
+	order: number,
+	props: Fill.Props
+) => {
+	if ( typeof children === 'function' ) {
+		return cloneElement( children( props ), { order } );
+	} else if ( isValidElement( children ) ) {
+		return cloneElement( children, { ...props, order } );
+	}
+	throw Error( 'Invalid children type' );
+};
+
+/**
+ * Sort fills by order for slot children.
+ *
+ * @param {Array} fills - slot's `Fill`s.
+ * @return {Node} Node.
+ */
+const sortFillsByOrder: Slot.Props[ 'children' ] = ( fills ) => {
```

Comment:
Since we've also used this method in the `header` should we create a shared utils file?

Reply:
Yeah re-using it would probably make the most sense. All of these do have to be moved to a package at some point.

Reply:
Addressed here: https://github.com/woocommerce/woocommerce/pull/34113/commits/c2a2760fa2be3fad95c6a4cd8eefc2fa1b1fa95e